src/exercises/day-1/luhn.rs

```diff
@@ -16,31 +16,32 @@
 // ANCHOR: luhn
 pub fn luhn(cc_number: &str) -> bool {
     // ANCHOR_END: luhn
+    let cc_number = cc_number.replace(" ", "");
     let mut sum = 0;
-    let mut double = false;
-    let mut digit_seen = 0;
+    let mut even = false;
```

Comment:
I think what you're essentially doing here is trading an allocation (`cc_number.replace(" ", "")` will typically have to allocate) for the `digit_seen` variable which tracks the number of characters seen.

I would prefer to avoid the allocation and thus remove the call to `replace` — allocation is not necessary when we can just filter out the spaces.

Your PR is useful since it can refactor the nested matching and if-statements into something that uses an early return.

Reply:
Can you take out this part since the allocation is ultimately unnecessary?